test/test_autograd.py

```diff
@@ -9605,9 +9605,10 @@ def pack(x):
             def unpack(name):
                 return torch.load(name)
 
+            a = torch.ones(5, requires_grad=True)
+            y = a * a
```

Comment:
you shouldnt need to do this? @bdhirsh

Reply:
Thanks for the review. Indeed I do not fully understand the effects. 
@bdhirsh This is what I was able to infer:
In test_saving_variable_to_disk defining tensors inside the saved_tensors_hooks context caused unintended side effects during assertions, specifically when calling _compare_regular_values_close. This was fixed by moving the tensor definitions (a and y) outside the saved_tensors_hooks context.

Reply:
Hi @bdhirsh would appreciate any guide/comment on this one!

Reply:
pulled your PR locally and this test fails for me with the change. taking a look